### PR TITLE
Type annotate the client

### PR DESCRIPTION
**Pull request by bowlofeggs**
Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

### CODE REVIEW

File: bodhi/client/__init__.py [hunk 1]
```diff
@@ -65,17 +68,19 @@ def _warn_if_url_or_openid_and_staging_set(ctx, param, value):
...
 
     Args:
-        ctx (click.core.Context): The Click context. Unused.
-        param (click.core.Option): The option being handled. Unused.
-        value (bool): The value of the --debug flag.
+        ctx: The Click context. Unused.
+        param: The option being handled. Unused.
+        value: The value of the --debug flag.
     Returns:
         bool: The value of the --debug flag.
```
Comment (sebwoj) [inline bodhi/client/__init__.py]: Let's remove bool.

File: bodhi/client/__init__.py [hunk 2]
```diff
@@ -261,18 +267,18 @@ def wrapper(*args, **kwargs):
...
-        user (unicode): The username to authenticate as.
-        password (unicode): The user's password.
-        staging (bool): Whether to use the staging server or not.
-        edit (bool): Set to True to edit an existing buildroot override.
-        kwargs (dict): Other keyword arguments passed to us by click.
+        url: The URL of a Bodhi server to create the update on. Ignored if staging is True.
+        user: The username to authenticate as.
+        password: The user's password.
+        staging: Whether to use the staging server or not.
+        edit: Set to True to edit an existing buildroot override.
```
Comment (sebwoj) [inline bodhi/client/__init__.py:283]: `openid_api` missing

File: bodhi/client/__init__.py [hunk 3]
```diff
@@ -437,17 +441,19 @@ def new(user, password, url, debug, openid_api, **kwargs):
...
+        value: str) -> str:
     """
     Validate the update argument given to the updates edit command.
 
     The update argument can only be an update id.
 
     Args:
-        param (basestring): The name of the parameter being validated. Unused.
-        value (basestring): The value of the value being validated.
+        param: The name of the parameter being validated. Unused.
```
Comment (sebwoj) [inline bodhi/client/__init__.py:459]: `ctx` is missing

File: bodhi/client/__init__.py [hunk 4]
```diff
@@ -587,11 +592,11 @@ def query(url, debug, mine=False, rows=None, **kwargs):
...
 
     Args:
-        url (unicode): The URL of a Bodhi server to create the update on. Ignored if staging is
+        url: The URL of a Bodhi server to create the update on. Ignored if staging is
                        True.
-        mine (Boolean): If the --mine flag was set
-        debug (Boolean): If the --debug flag was set
-        kwargs (dict): Other keyword arguments passed to us by click.
+        mine: If the --mine flag was set
+        debug: If the --debug flag was set
```
Comment (sebwoj) [inline bodhi/client/__init__.py:603]: `rows` is missing + what do you think about changing order of args in `Args` section to match the order of args in function?

File: bodhi/client/__init__.py [hunk 5]
```diff
@@ -897,28 +899,29 @@ def overrides():
...
-        url (unicode): The URL of a Bodhi server to create the update on. Ignored if staging is
-                       True.
-        packages (unicode): If supplied, the overrides for these package are queried
-        expired (bool): If supplied, True returns only expired overrides, False only active.
-        releases (unicode): If supplied, the overrides for these releases are queried.
-        builds (unicode): If supplied, the overrides for these builds are queried.
-        rows (unicode): The limit of rows displayed per page for query result.
-        page (unicode): If supplied, returns the results for a specific page number.
-        kwargs (dict): Other keyword arguments passed to us by click.
+        user: If supplied, overrides for this user will be queried.
```
Comment (sebwoj) [inline bodhi/client/__init__.py:931]: The same problem with args order here.

**Review by sebwoj — CHANGES_REQUESTED** [review]
Only minor problems.

File: bodhi/client/__init__.py (hunk 1 above)
Comment (bowlofeggs) [reply bodhi/client/__init__.py]: I'm applying this patch:

```
diff --git a/bodhi/client/__init__.py b/bodhi/client/__init__.py
index 8004f065..549d20ae 100644
--- a/bodhi/client/__init__.py
+++ b/bodhi/client/__init__.py
@@ -82,7 +82,7 @@ def _set_logging_debug(
         param: The option being handled. Unused.
         value: The value of the --debug flag.
     Returns:
-        bool: The value of the --debug flag.
+        The value of the --debug flag.
     """
     if value:
         ch = logging.StreamHandler()
```

File: bodhi/client/__init__.py (hunk 2 above)
Comment (bowlofeggs) [reply bodhi/client/__init__.py:283]: For this and the other missing things, I will make a second commit - teeeechnically they aren't really part of switching to type annotation, but I agree they should be documented.

File: bodhi/client/__init__.py (hunk 4 above)
Comment (bowlofeggs) [reply bodhi/client/__init__.py:603]: I'll do a third commit to switch the arg order to match.

**Review by puiterwijk — APPROVED** [review]
LGTM

**Comment by Zlopez** [comment]
Resolved conflict

**Pull request merged**